lib/adminware/config.rb

```diff
@@ -0,0 +1,64 @@
+require 'yaml'
+require 'fileutils'
+
+module Adminware
+  class Config
+    #Set default values for config settings
+    DEFAULT_JOBDIR = "jobs/"
+    DEFAULT_LOGFILE = "logs/adminware.log"
+    DEFAULT_STATEDIR = "var/state.yaml"
+
+    #Load the config file
+    def initialize
+      @path = Adminware.root 
+      configfile = File.join(@path, 'etc/config.yaml')
+      @config = YAML.load_file(configfile)
+
+      #If no path given in config then set them to their defaults
+      @jobdir = @config['centraljobdir'] ||= DEFAULT_JOBDIR
+      @jobdir = set_path(@jobdir)
+
+      @logfile = @config['logfile'] ||= DEFAULT_LOGFILE
+      @logfile = set_path(@logfile)
+
+      @statedir = @config['statedir'] ||= DEFAULT_STATEDIR
+      @statedir = set_path(@statedir)
+    end
+
+    #These methods return their respective files
+    def jobdir
+      @jobdir
+    end
+
+    def logfile
+      file_exist?(@logfile)
+      @logfile
+    end
+
+    def statedir
+      @statedir
+    end
+
+    private
+
+    #Check if a file exists
+    def file_exist?(file)
```

Comment:
This seems not to be a boolean check, but have the side effect of creating a file if it doesn't yet exist -- this method likely isn't named correctly.